astropy/io/fits/scripts/fitsheader.py

```diff
@@ -275,9 +292,71 @@ def print_headers_as_table(args):
     resulting_table.write(sys.stdout, format=args.table)
 
 
+def print_headers_as_comparison(args):
+    """Prints FITS header(s) with keywords as columns.
+
+    This follows the dfits+fitsort format.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments passed from the command-line as defined below.
+    """
+    from .... import table
+    tables = []
+    # Create a Table object for each file
+    for filename in args.filename:  # Support wildcards
+        try:
+            formatter = TableHeaderFormatter(filename)
+            tbl = formatter.parse(args.extensions,
+                                  args.keywords,
+                                  args.compressed)
+            tbl = tbl[np.where(tbl['keyword']!='')]  # Remove empty keywords
```

Comment:
`np.where` is probably useless here (I like to track the *Useless Use of np.where* :smile: ).

Reply:
If you run `fitsheader astropy/io/fits/tests/data/test0.fits`, you will see keyword-less lines. We need to get rid of these. They could even cause a crash due to duplicated columns (two empty keywords in the same fits header).

Reply:
Yes sure, sorry for the confusion, I meant that `np.where` is useless, not the filtering. You can use `tbl[tbl['keyword'] != '']`.